### PR TITLE
tools/functional-tester: make it work basically

### PR DESCRIPTION
**Pull request by yichengq**

### CODE REVIEW

File: tools/functional-tester/etcd-tester/failure_no.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+type failureNo struct {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/failure_no.go]: type baseFailure struct

File: tools/functional-tester/etcd-tester/tester.go [hunk 2]
```diff
@@ -14,41 +14,51 @@
...
 
 func (tt *tester) runLoop() {
 	for i := 0; i < tt.limit; i++ {
 		for j, f := range tt.failures {
-			fmt.Println("etcd-tester: [round#%d case#%d] start failure %s", i, j, f.Desc())
-			fmt.Println("etcd-tester: [round#%d case#%d] start injecting failure...", i, j)
-			if err := f.Inject(tt.agents); err != nil {
-				fmt.Println("etcd-tester: [round#%d case#%d] injection failing...", i, j)
-				tt.cleanup(i, j)
+			if err := tt.cluster.WaitFullHealth(); err != nil {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: Remove Full -> `WaitHealth`

File: tools/functional-tester/etcd-tester/failure_no.go (hunk 1 above)
Comment (xiang90) [inline tools/functional-tester/etcd-tester/failure_no.go]: or failureBase

File: tools/functional-tester/etcd-tester/failure_killall.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+package main
+
+type failureKillAll struct {
+	description
+}
+
+func newFailureKillAll() *failureKillAll {
+	return &failureKillAll{
+		description: "kill all members and restart them",
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/failure_killall.go]: remove `restart them`. it is the recovery phase. not the failure. this description describes the failure.

File: tools/functional-tester/etcd-agent/agent.go [hunk 4]
```diff
@@ -34,12 +35,20 @@ func newAgent(etcd string) (*Agent, error) {
 	}
 
 	c := exec.Command(etcd)
-	return &Agent{cmd: c}, nil
+
+	f, err := os.Create("etcd-run.log")
```
Comment (xiang90) [inline tools/functional-tester/etcd-agent/agent.go]: etcd.log?

Comment (yichengq) [inline tools/functional-tester/etcd-agent/agent.go]: both are fine to me. will change.

**Comment by yichengq** [comment]
comments all addressed.

File: tools/functional-tester/etcd-agent/agent.go [hunk 5]
```diff
@@ -23,6 +23,7 @@ import (
 
 type Agent struct {
 	cmd *exec.Cmd
+	f   *os.File
```
Comment (xiang90) [inline tools/functional-tester/etcd-agent/agent.go]: logfile *os.File

**Comment by xiang90** [comment]
LGTM

**Pull request merged**